extensions/MozChangeField/lib/BugStatus.pm

```diff
@@ -0,0 +1,57 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
Rather than splitting checks by field, it would be clearer and much more maintainable to split by function.

eg. one for the `canconfirm` checks, one for restrictions around reopening, etc